### DISCUSSION TITLE
Testing epishiny

### DISCUSSION THREAD

**Discussion by adamkucharski** [WHO-Collaboratory/collaboratory-epipipeline-community#9] [Ideas]
# Key links
Link to [GitHub repository](https://github.com/epicentre-msf/epishiny)

# Dependencies

```
# install.packages("remotes")
remotes::install_github("epicentre-msf/epishiny")
```

Need to install pkgload (not declared correctly as a dep)

# Initial problems identified

Package documentation not fully loaded (e.g. via roxygen2) because can’t find key functions after installing from GitHub.

Potential reason: don’t have `@export` tag in all functions. Maybe need devtools check?

# Work around

1. Clone repository
2. Source files from set up

```
library(epishiny)

# example package data
data("df_ll")
data("sf_yem")

# etc...
```

3. Load functions locally

```
file_path <- "~/Documents/GitHub/epishiny/R/"
source(paste0(file_path,"01_time.R"))
source(paste0(file_path,"02_place.R"))
source(paste0(file_path,"03_person.R"))
source(paste0(file_path,"04_filter.R"))
source(paste0(file_path,"05_launch.R"))
source(paste0(file_path,"utils.R"))
source(paste0(file_path,"utils.echart.R"))
```
4. Run test version
```
launch_demo_dashboard()
```

Date filters work. Group filters seem to have longer to load.

# Potential solutions for day 2:

1. Load locally and focus on epidemiological features. Some potential options:
- additional modules (R estimation, [cfr package](https://epiverse-trace.github.io/cfr/) etc.)
- alternative line list data (e.g. from [cfr case studies](https://github.com/CarmenTamayo/Applications-Epiverse-pipelines/blob/main/Marburg_underreporting.Rmd)
- feedback on visualisation options and other feedback.
- explore transmission chain module to examine use of multidimensional data (e.g. via existing [highcharter](https://jkunst.com/highcharter/) functionality).

2. Update code so correctly loads as package. But perhaps better for Paul at Epicentre to have first pass at this, unless quick and obvious fix.

**Comment by henryls1** [comment]
Potential solutions could include landscaping of future module developments

**Comment by henryls1** [comment]
###Other data sources can include: 

1. [global.health EVD Uganda (2022)](https://3mmuwilir3.execute-api.eu-central-1.amazonaws.com/web/url?folder=&file_name=latest.csv)
2. [global.health MVD Equatiorial Guinea (2023)](https://marburg-gh.s3.eu-central-1.amazonaws.com/latest.csv)

**Reply by Mougk** [reply]
Can also add [mpox line list](https://7rydd2v2ra.execute-api.eu-central-1.amazonaws.com/web/)

**Comment by adamkucharski** [comment]
We have summarised activites in the ‘for review’ issues here: https://github.com/epiverse-trace/epishiny/issues (we were working off fork for ease of sharing, but will move key issues/PRs to main repo once decided).

Good ones to start with to get idea of app structure would be:
- Document structure of app
- Test app on other datasets
- Document new modules

Then 'Add tables' and 'include options' summarise specific functionality that is going to be submitted as draft PR

**Reply by JamesABaker** [reply]
Thanks! Looks like there has been a lot of work going on! 🐝

I've added some comments to the issues you suggested, but this is obviously just a cursory glance at what looks like an incredibly complex problem! Please feel free to question and challenge what I have said 😄

**Comment by adamkucharski** [comment]
Thanks for the feedback. Summary of output PRs here:
https://github.com/epicentre-msf/epishiny/discussions/9#discussioncomment-7069922

**Comment by ntncmch** [comment]
For those interested in being involved in the development of epishiny, please add a 👍 to this post so I can add you to the meeting I'll organise with Paul Campbell (@PaulC91), our R Shiny dev, in the coming week or two.
